game/src/game/player.rs

```diff
@@ -61,6 +132,563 @@ fn setup_player(q: Query<(&Transform, Entity), Added<PlayerBlueprint>>, mut comm
             },
             animation: Default::default(),
         },));
+        println!("player spawned")
+    }
+}
+
+struct PlayerStatePlugin;
+
+impl Plugin for PlayerStatePlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            Update,
+            (
+                // test_trans_int,
+                // transition::<Idle, Grounded>,
+                transition::<Falling, Grounded>,
+                transition::<Grounded, Falling>,
+                // transition::<Falling, Jumping>,
+                transition::<Falling, Running>,
+                transition::<Grounded, Jumping>,
+                transition::<Jumping, Falling>,
+                transition::<Idle, Running>,
+                transition::<Running, Idle>,
+                apply_deferred,
+            ),
+        );
+        // add state systems
+        // add state transition systems
+    }
+}
+
+struct PlayerAnimationPlugin;
+
+impl Plugin for PlayerAnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            Update,
+            (
+                player_idle_animation,
+                player_falling_animation,
+                player_jumping_animation,
+                player_running_animation,
+            ),
+        );
+    }
+}
+
```

Comment:
I want a bit more time to think about your design/approach for player states. I will not review your player state abstraction in great detail right now. Expect another review with my thoughts on that.

Reply:
Some things I was trying to hit on with this design was a component based state machine, with concurrent states that could either interact with one another or not. I also want to be able to keep the states on the gameplay entity and not have to use children for concurrent states. I'm still considering whether it will be useful to write systems that query for/act on the PlayerStateTransition<C, N> components themselves.
Also open to talk through the design more if you want or if you have other ideas.